x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/transform/remove.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClientContract } from 'kibana/server';
+import { CallESAsCurrentUser, ElasticsearchAssetType, EsAssetReference } from '../../../../types';
+import { PACKAGES_SAVED_OBJECT_TYPE } from '../../../../../common/constants';
+
+export const stopTransforms = async (transformIds: string[], callCluster: CallESAsCurrentUser) => {
+  for (const transformId of transformIds) {
+    await callCluster('transport.request', {
+      method: 'POST',
+      path: `_transform/${transformId}/_stop`,
+      query: 'force=true',
+      ignore: [404],
+    });
+  }
+};
+
+export const deleteTransforms = async (
+  callCluster: CallESAsCurrentUser,
+  transformIds: string[]
+) => {
+  for (const transformId of transformIds) {
+    await stopTransforms([transformId], callCluster);
+    await callCluster('transport.request', {
+      method: 'DELETE',
+      query: 'force=true',
+      path: `_transform/${transformId}`,
+      ignore: [404],
+    });
+  }
+};
+
+export const deleteTransformRefs = async (
+  savedObjectsClient: SavedObjectsClientContract,
+  installedEsAssets: EsAssetReference[],
+  pkgName: string,
+  installedEsIdToRemove: string[],
+  currentInstalledEsTransformIds: string[]
+) => {
+  const filteredAssets = installedEsAssets.filter(({ type, id }) => {
+    if (type !== ElasticsearchAssetType.transform) return true;
+    return currentInstalledEsTransformIds.includes(id) || !installedEsIdToRemove.includes(id);
```

Comment:
I notice this would result in a bug for integrations  being reinstalled because they'd have the same name and version resulting in duplicate references, which would result in many requests being made for each reinstall as it tries to stop and delete the same transform x times.  The same issue exists for pipelines. Somewhat of an edgecase, but should be fixed.  I was thinking we could do something like checking that values are unique before returning filtered assets like `uniqBy(filteredAssets, 'id)` (from lodash) for now.  I can follow up with a fix for pipelines.